ui/components/FilterDialog.tsx

```diff
@@ -138,8 +138,8 @@ type sectionSelectObject = { [header: string]: boolean };
 function UnstyledFilterDialog({
```

Comment:
I know it was not added in this PR, but can we just remove `=== true` on line 55 of `FilterDialog`? Lines 54+55 can be replaced with:

```
compoundKeys.length > 0 && compoundKeys.every((key) => formState[key]);
```

Type of formState is already `{ [inputName: string]: boolean }`

Reply:
BTW, I'm getting an unrelated warning ` Warning: An update to CoreClientContextProvider inside a test was not wrapped in act(`, can you please re-check the UI tests after rebase?

Reply:
That error in the UI tests doesn't give a lot of info about where it's coming from besides that it pops up in the terminal right after the Breadcrumbs tests and before the Footer tests. I ran DataTable and DataTableFilter tests just to make sure it wasn't there, then I went into Breadcrumbs and made a small change in the context to match other tests. It's possible that the `createMockClient({})` was causing the issue because changing it to just `{}` fixes the error. The tests still pass but I might have broken something you were doing so let me know if that's a terrible change 🌞

Reply:
Looking into it now.

There was another warning

```
      TypeError: api.GetFeatureFlags is not a function
          at Object.FeatureFlags.react_query_1.useQuery.staleTime [as queryFn] (/Volumes/Macintosh_HD_Data/_job_ww/_src/weave-gitops/ui/contexts/CoreClientContext.tsx:51:15)
```
while running the UI tests.
@ozamosi is it OK to add typeof Core to this function `function FeatureFlags(api: typeof Core) {` ?

If I do it, this warning disappears.

Reply:
Checked it: wrapping everything with `act` and `async` (with switching to `render` instead of `renderer`) does not fix the `GetFeatureFlags` warnings (new warnings appear in other components). And I see the `not wrapped with act` warnings in the `main` branch. It looks like these test issues are not directly related to the current PR. We need to ask @ozamosi on Tuesday what can be done about it and which mocks are required and where.